dbt/adapters/impala/connections.py

```diff
@@ -159,27 +160,58 @@ def open(cls, connection):
                     port=credentials.port,
                     retries=credentials.retries,
                 )
+            connection_end_time = time.time()
 
-            connection.state = "open"
+            connection.state = ConnectionState.OPEN
             connection.handle = handle
         except Exception as ex:
             LOGGER.debug("Connection error {}".format(ex))
-            connection.state = "fail"
+            connection.state = ConnectionState.FAIL
             connection.handle = None
+            connection_end_time = time.time()
             pass
 
         # track usage
         payload = {
             "id": "dbt_impala_open",
-            "unique_hash": hashlib.md5(credentials.host.encode()).hexdigest(),
+            "unique_hash": hashlib.md5((str(credentials.host) + str(credentials.username)).encode()).hexdigest(),
             "auth": auth_type,
+            "connection_name": connection.name,
             "connection_state": connection.state,
+            "elapsed_time": "{:.2f}".format(connection_end_time - connection_start_time),
         }
 
         tracker.track_usage(payload)
 
         return connection
 
+    @classmethod
+    def close(cls, connection):
+        try:
+            # if the connection is in closed or init, there's nothing to do
+            if connection.state in {ConnectionState.CLOSED, ConnectionState.INIT}:
+                return connection
+
+            connection_close_start_time = time.time()
+            connection = super().close(connection)
+            connection_close_end_time = time.time()
+
+            credentials = connection.credentials
+
+            payload = {
+                "id": "dbt_impala_close",
+                "unique_hash": hashlib.md5((str(credentials.host) + str(credentials.username)).encode()).hexdigest(),
+                "connection_name": connection.name,
+                "connection_state": ConnectionState.CLOSED,
+                "elapsed_time": "{:.2f}".format(connection_close_end_time - connection_close_start_time),
```

Comment:
nitpick connection_duration or duration_of_connection maybe

Reply:
or does this just mean how much time it took to close a connection. in that case that is also not useful.

Reply:
it is the time it took to close the connection, not connection_duration.

Reply:
Ok understood. Yeah got confused as i don't know if its useful to track how much time a connection takes to open or connection takes to close. usually should be in ms

On the other hand , connection duration might give info on how long a connection was kept open and used.

Reply:
connection_duration will need to be separately tracked, this is simply the time it takes to open or close a connection. For impala it is usually in ms, but sometimes can be in seconds depending on the server being used. For other adapters however, this time could be signification, so I think it is good to have metrics.